src/azure-firewall/azext_firewall/custom.py

```diff
@@ -850,18 +746,21 @@ def add_firewall_policy_intrusion_detection_config(cmd,
         raise RequiredArgumentMissingError('Intrusion detection mode is not set. Setting it by update command first')
 
     if signature_id is not None and signature_mode is not None:
-        for overrided_signature in firewall_policy.intrusion_detection.configuration.signature_overrides:
-            if overrided_signature.id == signature_id:
-                raise InvalidArgumentValueError(
-                    f'Signature ID {signature_id} exists. Delete it first or try update instead')
-
         FirewallPolicyIntrusionDetectionSignatureSpecification = \
             cmd.get_models('FirewallPolicyIntrusionDetectionSignatureSpecification')
         signature_override = FirewallPolicyIntrusionDetectionSignatureSpecification(
             id=signature_id,
             mode=signature_mode
         )
-        firewall_policy.intrusion_detection.configuration.signature_overrides.append(signature_override)
+        if firewall_policy.intrusion_detection.configuration is not None:
+            for overrided_signature in firewall_policy.intrusion_detection.configuration.signature_overrides:
+                if overrided_signature.id == signature_id:
+                    raise InvalidArgumentValueError(
+                        f'Signature ID {signature_id} exists. Delete it first or try update instead')
+            firewall_policy.intrusion_detection.configuration.signature_overrides.append(signature_override)
+        else:
+            Configuration = cmd.get_models('FirewallPolicyIntrusionDetectionConfiguration')
+            firewall_policy.intrusion_detection.configuration = Configuration(signature_overrides=[signature_override])
```

Comment:
Could we decouple with SDK here?

Reply:
This requires migrating the ‘az network firewall policy ’subcommand, which is not currently done. If decoupling is needed, I can migrate the subcommands.